controllers/provider-alicloud/pkg/apis/alicloud/helper/helper.go

```diff
@@ -47,11 +47,30 @@ func FindSecurityGroupByPurpose(securityGroups []alicloud.SecurityGroup, purpose
 // FindMachineImage takes a list of machine images and tries to find the first entry
 // whose name and version matches with the given name and version. If no such entry is
 // found then an error will be returned.
-func FindMachineImage(machineImages []alicloud.MachineImage, name, version string) (*alicloud.MachineImage, error) {
-	for _, machineImage := range machineImages {
-		if machineImage.Name == name && machineImage.Version == version {
+func FindMachineImage(configImages []api.MachineImage, imageName, imageVersion string) (*api.MachineImage, error) {
+	for _, machineImage := range configImages {
+		if machineImage.Name == imageName && machineImage.Version == imageVersion {
 			return &machineImage, nil
 		}
 	}
-	return nil, fmt.Errorf("no machine image name %q in version %q found", name, version)
+	return nil, fmt.Errorf("no machine image name %q in version %q found", imageName, imageVersion)
+}
+
+// FindImageForRegionFromCloudProfile takes a list of machine images, and the desired image name, version, and region. It tries
+// to find the image with the given name and version in the desired region. If it cannot be found then an error
+// is returned.
+func FindImageForRegionFromCloudProfile(profileConfig *api.CloudProfileConfig, imageName, imageVersion string) (string, error) {
```

Comment:
what is the reasoning behind `ForRegion` in the name?

Reply:
You are right, this makes no sense here. I've removed it.